libs/MeshKernelApi/tests/src/ApiTest.cpp

```diff
@@ -1522,6 +1562,53 @@ TEST_F(CartesianApiTestFixture, CountNodesInPolygonMesh2D_OnMesh2D_ShouldCountAl
     ASSERT_EQ(12, numNodes);
 }
 
+TEST_F(CartesianApiTestFixture, GetNodesInPolygonMesh2D_OnMesh2D_NodeInPolygonFailures)
```

Comment:
can you put some of the cache tests in a separate file, such as ApiCacheTest? ApiTest is already too large and we should split it in part